src/main/java/datawave/microservice/query/messaging/kafka/KafkaQueryResultsManager.java

```diff
@@ -0,0 +1,200 @@
+package datawave.microservice.query.messaging.kafka;
+
+import static datawave.microservice.query.messaging.kafka.KafkaQueryResultsManager.KAFKA;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DeleteRecordsResult;
+import org.apache.kafka.clients.admin.RecordsToDelete;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.TopicPartitionInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.kafka.config.TopicBuilder;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.KafkaAdmin;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.core.ProducerFactory;
+import org.springframework.stereotype.Component;
+
+import datawave.microservice.query.messaging.QueryResultsListener;
+import datawave.microservice.query.messaging.QueryResultsManager;
+import datawave.microservice.query.messaging.QueryResultsPublisher;
+import datawave.microservice.query.messaging.config.MessagingProperties;
+
+@Component
+@ConditionalOnProperty(name = "datawave.query.messaging.backend", havingValue = KAFKA)
+public class KafkaQueryResultsManager implements QueryResultsManager {
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    
+    public static final String KAFKA = "kafka";
+    
+    private final MessagingProperties messagingProperties;
+    private final AdminClient adminClient;
+    private final ProducerFactory<String,String> kafkaProducerFactory;
+    private final ConsumerFactory<String,String> kafkaConsumerFactory;
+    
+    public KafkaQueryResultsManager(MessagingProperties messagingProperties, KafkaAdmin adminClient, ProducerFactory<String,String> kafkaProducerFactory,
+                    ConsumerFactory<String,String> kafkaConsumerFactory) {
+        this.messagingProperties = messagingProperties;
+        this.adminClient = AdminClient.create(adminClient.getConfigurationProperties());
+        this.kafkaProducerFactory = kafkaProducerFactory;
+        this.kafkaConsumerFactory = kafkaConsumerFactory;
+    }
+    
+    /**
+     * Create a listener for a specified listener id
+     *
+     * @param listenerId
+     *            The listener id
+     * @return a query queue listener
+     */
+    @Override
+    public QueryResultsListener createListener(String listenerId, String queryId) {
+        createTopic(queryId);
```

Comment:
In Rabbit we use a QUERY_QUEUE_PREFIX. Should we also use a QUERY_TOPIC_PREFIX here?

Reply:
I think the intent was for both the queues and topics to just be named after the query id.  let me take a look at this and get back to you.

Reply:
That was the idea, using the query id in both names.

Reply:
ok, i updated all of the results messaging subsystems to use the prefix.  all tested and working.